src/DIRAC/DataManagementSystem/Client/FTS3Job.py

```diff
@@ -306,138 +323,154 @@ def _constructTransferJob(self, pinTime, allLFNs, target_spacetoken, protocols=N
+                # at the job construction step in FTS3Operation).
+                # This test is important, because multiple files would result in the source
+                # being deleted !
+                if len(allLFNs) != 1:
+                    log.debug("Multihop job has %s files while only 1 allowed" % len(allLFNs))
+                    return S_ERROR(errno.E2BIG, "Trying multihop job with more than one file !")
+
+                res = srcSE.getURL(allSrcDstSURLs, protocol=srcSE.localStageProtocolList)
+
+                if not res["OK"]:
+                    return res
+
+                for lfn, reason in res["Value"]["Failed"].items():
+                    failedLFNs.add(lfn)
+                    log.error("Could not get stage SURL", "%s %s" % (lfn, reason))
+                    allSrcDstSURLs.pop(lfn)
+
+                allStageURLs = res["Value"]["Successful"]
 
-        transfers = []
-
-        fileIDsInTheJob = []
-
-        for ftsFile in self.filesToSubmit:
-
-            if ftsFile.lfn in failedLFNs:
-                log.debug("Not preparing transfer for file %s" % ftsFile.lfn)
-                continue
-
-            sourceSURL, targetSURL = allSrcDstSURLs[ftsFile.lfn]
-            stageURL = allStageURLs.get(ftsFile.lfn)
-
-            if sourceSURL == targetSURL:
-                log.error("sourceSURL equals to targetSURL", "%s" % ftsFile.lfn)
-                ftsFile.error = "sourceSURL equals to targetSURL"
-                ftsFile.status = "Defunct"
-                continue
-
-            ftsFileID = getattr(ftsFile, "fileID")
-
-            # Under normal circumstances, we simply submit an fts transfer as such:
-            # * srcProto://myFile -> destProto://myFile
-            #
-            # Even in case of the source storage being a tape system, it works fine.
-            # However, if the staging and transfer protocols are different (which might be the case for CTA),
-            #  we use the multihop machinery to submit two sequential fts transfers:
-            # one to stage, one to transfer.
-            # It looks like such
-            # * stageProto://myFile -> stageProto://myFile
-            # * srcProto://myFile -> destProto://myFile
+            for ftsFile in self.filesToSubmit:
 
-            if stageURL:
+                if ftsFile.lfn in failedLFNs:
+                    log.debug("Not preparing transfer for file %s" % ftsFile.lfn)
+                    continue
+
+                sourceSURL, targetSURL = allSrcDstSURLs[ftsFile.lfn]
+                stageURL = allStageURLs.get(ftsFile.lfn)
+
+                if sourceSURL == targetSURL:
+                    log.error("sourceSURL equals to targetSURL", "%s" % ftsFile.lfn)
+                    ftsFile.error = "sourceSURL equals to targetSURL"
+                    ftsFile.status = "Defunct"
+                    continue
+
+                ftsFileID = getattr(ftsFile, "fileID")
```

Comment:
It's because of sqlalchemy. We should not define `fileID` ourselves, so then pylint does not find it. So it would be a real problem if the attribute does not exist, but I write it this way to have pylint happy